backend/controllers/passController.js

```diff
@@ -55,14 +62,47 @@ const viewPasses = asyncHandler(async (req, res) => {
         message: "Passes fetched successfully",
       })
     } else if (passes.length < 1) {
-      res.status(201).json({
-        success: true,
-        message: "No passes found",
-      })
+      throw new Error("No passes found")
     }
   } catch (err) {
     console.log(err)
+    res.status(400).json(err?.message)
+  }
+})
+
+//@desc Verify pass by gaurd
+//@route PUT /users/verifyPass
+const verifyPass = asyncHandler(async (req, res) => {
+  try {
+    await connectDB()
+    const user = await User.findById(req.user._id)
+    console.log(req.params._id)
+
+    if (user.role == "gaurd") {
+      const passVerified = await Pass.findOneAndUpdate(
+        { _id: req.params._id },
+        {
+          checkInStatus: true,
+          timestamps: new Date(),
+        }
+      )
+      console.log(passVerified)
+      if (passVerified) {
```

Comment:
Why this condition is required?

Reply:
if in case pass verification fails due to any technical issue or if in any case checkInStatus and time couldn't be updated, i thought this should be put, but not sure now (since you asked..)

Reply:
Try the simulate the scenario and then let me know? Before that read about try and catch and error handling.

Reply:
since you mentioned try and catch and error handling, it is obvious that any error could be reverted with throwing error. So we don't need to put extra conditions to check from our end

Reply:
Great. So to conclude I could say that if passVerifcation gets any error, the compiler will directly go to catch block and skip all the following code. Reply with yes if you understood it.

Reply:
yes

Reply:
"koi bhi error ayega, catch block execute hoga seedha error display krne ko"

Reply:
good